spark/src/main/scala/geotrellis/spark/io/CollectionLayerReader.scala

```diff
@@ -0,0 +1,76 @@
+package geotrellis.spark.io
+
+import geotrellis.spark._
+import geotrellis.spark.io.avro._
+import geotrellis.spark.io.json._
+import geotrellis.util._
+
+import spray.json._
+
+import scala.reflect._
+
+abstract class CollectionLayerReader[ID] { self =>
+  val defaultNumPartitions = CollectionLayerReader.defaultNumPartitions
+
+  def read[
+    K: AvroRecordCodec: Boundable: JsonFormat: ClassTag,
+    V: AvroRecordCodec: ClassTag,
+    M: JsonFormat: GetComponent[?, Bounds[K]]
+  ](id: ID, rasterQuery: LayerQuery[K, M], numPartitions: Int, indexFilterOnly: Boolean): Seq[(K, V)] with Metadata[M]
```

Comment:
this read should not have `numPartitions` argument. All the reads should be using threads to distributed the work as much as makes sense, but by definition you only ever get one partition, the collection.